### PR TITLE
✨Sort career order

### PR DESCRIPTION
**Pull request by AndrewJDick**
Careers are now initially sorted alphabetically.
Once sorted, they are then ordered based on the career's frontmatter `order`, or a fallback if none exists. 

Result:
```
UX
  a
  b

Design 
  a
  b

[...]
  a
  b

etc
``

Closes: https://github.com/fathomlondon/fathomlondon.com/issues/275

### CODE REVIEW

**Comment by peduarte** [comment]
@AndrewJDick I think @saracmcorreia's requirement was to have the `tech` category first

**Comment by saracmcorreia** [comment]
@peduarte - I asked for the following order: UX -> UI -> Tech -> other. After all we are primarily a UX/UI consultancy ;)

**Comment by peduarte** [comment]
@AndrewJDick  What's the reason you chose to not use an indexed `order` field?

**Comment by AndrewJDick** [comment]
@peduarte refactored to use an indexed order field

File: src/pages/careers.js [hunk 1]
```diff
@@ -5,6 +5,24 @@ import { TitleAndMetaTags } from '../components/TitleAndMetaTags';
...
+function orderCareers(rawCareers) {
+	const defaultIndex = setDefaultIndex(rawCareers);
+
+	return rawCareers.sort((a, b) => {
+		const current = a.node.frontmatter.order || defaultIndex;
+		const next = b.node.frontmatter.order || defaultIndex;
+		return current < next ? -1 : current > next ? 1 : 0;
+	});
+}
+
```
Comment (peduarte) [inline src/pages/careers.js]: Do you think we could maybe rely on something like this instead:
https://lodash.com/docs/4.17.5#orderBy

Comment (peduarte) [reply src/pages/careers.js]: or maybe `sortBy`...

**Comment by peduarte** [comment]
Another thing: you need to make sure that this new field can be edited in the CMS. Let me know if you need help with this

**Comment by AndrewJDick** [comment]
@peduarte removed the order function as per your comment. 

I added the order field to the `config.yml`. The field appeared in each career page on the CMS admin panel, but it was not pulling the order value from the `.md` files (the field appeared blank). Any idea why that is?

File: static/admin/config.yml [hunk 2]
```diff
@@ -41,6 +41,7 @@ collections:
     folder: "content/careers"
     create: true
     fields:
+      - { label: "Order", name: "order", widget: "number", valueType: "int", min: 1 }
```
Comment (peduarte) [inline static/admin/config.yml]: why is min `1` instead of `0`?

Comment (AndrewJDick) [reply static/admin/config.yml]: @peduarte I just think it reads better; priority one is usually associated with top priority, whereas priority 0 sounds like no priority. 

What do you think?

Comment (peduarte) [reply static/admin/config.yml]: Ok, I think for now it's cool, but `0` could be helpful if we wanted to make a specific role appear first in the list.

The ordering rule would behave quite similarly to `flex order`. But we can change this later if we need to.

Will merge and let's see how it works ;)

**Pull request merged**